src/MBBSEmu.Btrieve/BtrieveFile.cs

```
@@ -444,34 +566,45 @@ private void LoadBtrieveRecords(ILogger logger)
                 for (var j = 0; j < recordsInPage; j++)
                 {
                     if (recordsLoaded == RecordCount)
-                        break;
+                        goto finished_loaded;
 
                     var recordOffset = (uint)pageOffset + (uint)(PhysicalRecordLength * j);
                     // Marked for deletion? Skip
                     if (DeletedRecordOffsets.Contains(recordOffset))
                         continue;
 
                     var record = Data.AsSpan().Slice((int)recordOffset, PhysicalRecordLength);
-                    if (IsUnusedRecord(record))
+                    if (IsUnusedRecord(record, v6))
                         break;
 
+                    recordOffset += dataOffset;
                     var recordArray = new byte[RecordLength];
                     Array.Copy(Data, recordOffset, recordArray, 0, RecordLength);
 
-                    if (VariableLengthRecords)
+                    try
                     {
-                        using var stream = new MemoryStream();
-                        stream.Write(recordArray);
-
-                        Records.Add(new BtrieveRecord(recordOffset, GetVariableLengthData(recordOffset, stream)));
+                        if (VariableLengthRecords)
+                        {
+                            using var stream = new MemoryStream();
+                            stream.Write(recordArray);
+
+                            Records.Add(new BtrieveRecord(recordOffset, GetVariableLengthData(recordOffset, stream, v6)));
+                        }
+                        else
+                            Records.Add(new BtrieveRecord(recordOffset, recordArray));
                     }
-                    else
-                        Records.Add(new BtrieveRecord(recordOffset, recordArray));
+                    catch (ArgumentException ex)
+                    {
+                        if (!allowCorruptedRecords)
+                            throw ex;
```

This should just be `throw` if you want to throw the exception that was caught